### PR TITLE
Tests: improve native path representation

### PR DESCRIPTION
**Pull request by compnerd**
Rather than hardcoding the paths and the path separator, we now compute
the actual locations to ensure that we can correctly represent the path
representation natively on the target.  This enables the correct
representation on Windows.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 1]
```diff
@@ -27,6 +27,14 @@ private func executableName(_ name: String) -> String {
 #endif
 }
 
+private func rebase(_ arc: String, at base: AbsolutePath) -> String {
```
Comment (compnerd) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:30]: If you have a better name for this function, please suggest it.  I'm not tied to it, but this was the best I could come up with.

**Comment by compnerd** [comment]
@cltnschlosser @artemcm - I'd like to get this merged as it unblocks another set of changes.

**Review by cltnschlosser — APPROVED** [review]
This change is needed because, Windows needs the drive letter for it to be a valid absolute path?

**Review by artemcm — APPROVED** [review]

**Comment by compnerd** [comment]
> This change is needed because, Windows needs the drive letter for it to be a valid absolute path?

Correct.  The Unix absolute path `/foo/bar` is interpreted as `\foo\bar` which is a drive relative path.  We compute the absolute path by taking the current drive and appending that relative path to it.  This gives us an absolute path.

**Pull request merged**